src/bridge/cockpitcpusamples.c

```diff
@@ -71,28 +71,45 @@ cockpit_cpu_samples (CockpitSamples *samples)
       guint64 system;
       guint64 idle;
       guint64 iowait;
+      gchar cpu_core[8];
+      gchar core[5];
 
-      if (!(g_str_has_prefix (line, "cpu ")))
+      if (!(g_str_has_prefix (line, "cpu")))
         continue;
 
       #define FMT64 "%" G_GUINT64_FORMAT " "
-      if (sscanf (line + sizeof ("cpu ") - 1, FMT64 FMT64 FMT64 FMT64 FMT64,
+      if (sscanf (line, "%s " FMT64 FMT64 FMT64 FMT64 FMT64,
```

Comment:
Let's please limit the string length, otherwise `cpu_core` could possibly overflow the stack. `%8s` should do the trick. To avoid duplicating the length, try something like
```c
#define CPU_CORE_MAXLEN 8
gchar cpu_core[CPU_CORE_MAXLEN]
sscanf(..., "%" #CPU_CORE_MAXLEN "s ..."
```

Reply:
For my own future self: This can be done with [G_STRINGIFY](https://developer.gnome.org/glib/stable/glib-Miscellaneous-Macros.html#G-STRINGIFY:CAPS). The `#` operator only works inside macros.

Reply:
```
#include <stdio.h>

int
main (void)
{
  char x[5] = "aaaa";

  sscanf ("abcd", "%3s", x);
  printf("%s\n", x);

  return 0;
}
```

```
toolbox:~$ ./x
abc
```

ie: `%3s` means "3 characters, plus nul" (ie: 4 characters).  Make sure to add 1 to the buffer to compensate.